src/marqo/core/embed/embed.py

```diff
@@ -0,0 +1,123 @@
+from timeit import default_timer as timer
+from typing import List, Optional, Union, Iterable, Sequence, Dict, Any, Tuple
+import numpy as np
+
+import marqo.core.unstructured_vespa_index.common as unstructured_common
+from marqo import marqo_docs
+from marqo import exceptions as base_exceptions
+from marqo.api import exceptions as api_exceptions
+from marqo.core import exceptions as core_exceptions
+# We depend on _httprequests.py for now, but this may be replaced in the future, as
+# _httprequests.py is designed for the client
+from marqo.core import constants
+
+from marqo.core.index_management.index_management import IndexManagement
+from marqo.core.models.marqo_index import IndexType
+from marqo.core.models.marqo_index import MarqoIndex, FieldType, UnstructuredMarqoIndex, StructuredMarqoIndex
+from marqo.core.models.marqo_query import MarqoTensorQuery, MarqoLexicalQuery
+from marqo.core.structured_vespa_index.structured_vespa_index import StructuredVespaIndex
+from marqo.core.unstructured_vespa_index import unstructured_validation as unstructured_index_add_doc_validation
+from marqo.core.unstructured_vespa_index.unstructured_vespa_index import UnstructuredVespaIndex
+from marqo.core.vespa_index import for_marqo_index as vespa_index_factory
+from marqo.s2_inference import errors as s2_inference_errors
+from marqo.s2_inference import s2_inference
+from marqo.s2_inference.clip_utils import _is_image
+from marqo.s2_inference.processing import image as image_processor
+from marqo.s2_inference.processing import text as text_processor
+from marqo.s2_inference.reranking import rerank
+from marqo.tensor_search import delete_docs
+from marqo.tensor_search import enums
+from marqo.tensor_search import index_meta_cache
+from marqo.tensor_search import utils, validation, add_docs, tensor_search
+from marqo.tensor_search.enums import (
+    Device, TensorField, SearchMethod, EnvVars
+)
+from marqo.tensor_search.index_meta_cache import get_cache, get_index
+from marqo.tensor_search.models.add_docs_objects import AddDocsParams
+from marqo.tensor_search.models.api_models import BulkSearchQueryEntity, ScoreModifier
+from marqo.tensor_search.models.delete_docs_objects import MqDeleteDocsRequest
+from marqo.tensor_search.models.private_models import ModelAuth
+from marqo.tensor_search.models.search import Qidx, JHash, SearchContext, VectorisedJobs, VectorisedJobPointer
+from marqo.tensor_search.telemetry import RequestMetricsStore
+from marqo.tensor_search.tensor_search_logging import get_logger
+from marqo.vespa.exceptions import VespaStatusError
+from marqo.vespa.models import VespaDocument, FeedBatchResponse, QueryResult
+
+logger = get_logger(__name__)
+
+
+class Embed:
+    def __init__(self, config):
+        self.config = config
```

Comment:
The config object is problematic. It has extra things you don't really depend on and tends to cause dependency cycles when core objects depend on each other. Replace with specific dependencies